src/components/compress/index.tsx

```diff
@@ -289,6 +292,19 @@ export default class Compress extends Component<Props, State> {
     document.title = filename ? `${filename} - ${originalDocumentTitle}` : originalDocumentTitle;
   }
 
+  private updateLoadingIndicator(isLoading: boolean = false): void {
+    // Update the title if the editor loading state changes
+    if (isLoading) {
+      if (!document.title.startsWith(loadingIndicator)) {
```

Comment:
This feels a bit fragile. Eg, if the `filename` starts with "⏳ " then this code is going to think there's a loading indicator.

Ideally we should avoid relying on reading state from `document.title`, and instead generate it from app state.